### ISSUE TITLE
Getting errors when installing Appium 1.11.0

### ISSUE THREAD

**Issue by BoazWarshawsky** [appium/appium#12181]
## The problem
I am getting errors during install of appium which later on fail my test to connect to appium server:

##[debug]Evaluating condition for step: 'Setup Appium server'
##[debug]Evaluating: SucceededNode()
##[debug]Evaluating SucceededNode:
##[debug]=> True
##[debug]Result: True
##[section]Starting: Setup Appium server
==============================================================================
Task         : Shell Script
Description  : Run a shell script using bash
Version      : 2.1.3
Author       : Microsoft Corporation
Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613738)
==============================================================================
##[debug]agent.workFolder=/Users/vsts/agent/2.146.0/work
##[debug]loading inputs and endpoints
##[debug]loading INPUT_SCRIPTPATH
##[debug]loading INPUT_DISABLEAUTOCWD
##[debug]loading INPUT_CWD
##[debug]loading INPUT_FAILONSTANDARDERROR
##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
##[debug]loading SECRET_SYSTEM_ACCESSTOKEN
##[debug]loaded 8
##[debug]check path : /Users/vsts/agent/2.146.0/work/_tasks/ShellScript_6c731c3c-3c68-459a-a5c9-bde6e6595b5b/2.1.3/task.json
##[debug]set resource file to: /Users/vsts/agent/2.146.0/work/_tasks/ShellScript_6c731c3c-3c68-459a-a5c9-bde6e6595b5b/2.1.3/task.json
##[debug]system.culture=en-US
##[debug]check path : /bin/bash
##[debug]bash=/bin/bash
##[debug]scriptPath=/Users/vsts/agent/2.146.0/work/1/s/ci_scripts/04_setup_appium_server.sh
##[debug]check path : /Users/vsts/agent/2.146.0/work/1/s/ci_scripts/04_setup_appium_server.sh
##[debug]cwd=/Users/vsts/agent/2.146.0/work/1/s
##[debug]cwd=/Users/vsts/agent/2.146.0/work/1/s
##[debug]Absolute path for pathSegments: /Users/vsts/agent/2.146.0/work/1/s = /Users/vsts/agent/2.146.0/work/1/s
##[debug]build.sourcesDirectory=/Users/vsts/agent/2.146.0/work/1/s
##[debug]Absolute path for pathSegments: /Users/vsts/agent/2.146.0/work/1/s = /Users/vsts/agent/2.146.0/work/1/s
##[debug]cwdpath supplied :false
##[debug]disableAutoCwd=false
##[debug]testing directory '/Users/vsts/agent/2.146.0/work/1/s/ci_scripts'
##[debug]/bin/bash arg: /Users/vsts/agent/2.146.0/work/1/s/ci_scripts/04_setup_appium_server.sh
##[debug]args=null
##[debug]failOnStandardError=false
##[debug]exec tool: /bin/bash
##[debug]Arguments:
##[debug]   /Users/vsts/agent/2.146.0/work/1/s/ci_scripts/04_setup_appium_server.sh
[command]/bin/bash /Users/vsts/agent/2.146.0/work/1/s/ci_scripts/04_setup_appium_server.sh
Install Appium 1.11.0‌
npm WARN deprecated fsevents@2.0.1: Not Ready For Production‌
/usr/local/bin/appium -> /usr/local/lib/node_modules/appium/build/lib/main.js

> fsevents@1.2.6 install /usr/local/lib/node_modules/appium/node_modules/chokidar/node_modules/fsevents
> node install

node-pre-gyp WARN Using request for node-pre-gyp https download 
[fsevents] Success: "/usr/local/lib/node_modules/appium/node_modules/chokidar/node_modules/fsevents/lib/binding/Release/node-v67-darwin-x64/fse.node" is installed via remote

> appium-chromedriver@4.10.1 install /usr/local/lib/node_modules/appium/node_modules/appium-chromedriver
> node install-npm.js

[05:17:56] [Chromedriver Install] Installing Chromedriver version '2.45' for platform 'mac' and architecture '64'
[05:17:56] [Chromedriver Install] Opening temp file to write 'chromedriver_mac64' to...
[05:17:56] [Chromedriver Install] Opened temp file '/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/2019118-858-jw26ab.pijyo/chromedriver_mac64.zip'
[05:17:56] [Chromedriver Install] Downloading https://chromedriver.storage.googleapis.com/2.45/chromedriver_mac64.zip...
[05:17:59] [Chromedriver Install] Writing binary content to /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/2019118-858-jw26ab.pijyo/chromedriver_mac64.zip...
[05:17:59] [Chromedriver Install] Extracting /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/2019118-858-jw26ab.pijyo/chromedriver_mac64.zip to /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/2019118-858-jw26ab.pijyo/chromedriver_mac64
[05:17:59] [Chromedriver Install] Creating /usr/local/lib/node_modules/appium/node_modules/appium-chromedriver/chromedriver/mac...
[05:17:59] [Chromedriver Install] Copying unzipped binary, reading from /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/2019118-858-jw26ab.pijyo/chromedriver_mac64/chromedriver...
[05:18:00] [Chromedriver Install] Writing to /usr/local/lib/node_modules/appium/node_modules/appium-chromedriver/chromedriver/mac/chromedriver...
[05:18:00] [Chromedriver Install] /usr/local/lib/node_modules/appium/node_modules/appium-chromedriver/chromedriver/mac/chromedriver successfully put in place

> appium-selendroid-driver@1.13.3 install /usr/local/lib/node_modules/appium/node_modules/appium-selendroid-driver
> node ./bin/install.js

[05:18:05] Java version 1.8.0_202 found
[05:18:05] Ensuring /usr/local/lib/node_modules/appium/node_modules/appium-selendroid-driver/selendroid/download exists
[05:18:05] Downloading Selendroid standalone server version 0.17.0 from https://repo1.maven.org/maven2/io/selendroid/selendroid-standalone/0.17.0/selendroid-standalone-0.17.0-with-dependencies.jar --> /usr/local/lib/node_modules/appium/node_modules/appium-selendroid-driver/selendroid/download/selendroid-server-7cf7163ac47f1c46eff95b62f78b58c1dabdec534acc6632da3784739f6e9d82.jar
[05:18:06] Writing binary content to /usr/local/lib/node_modules/appium/node_modules/appium-selendroid-driver/selendroid/download/selendroid-server.jar.tmp
[05:18:06] Selendroid standalone server downloaded
[05:18:06] Determining AndroidManifest location
[05:18:07] Determining server apk location
[05:18:07] Extracting manifest and apk to /usr/local/lib/node_modules/appium/node_modules/appium-selendroid-driver/selendroid/download
[05:18:08] Copying manifest and apk to /usr/local/lib/node_modules/appium/node_modules/appium-selendroid-driver/selendroid
[05:18:08] Cleaning up temp files
[05:18:08] Fixing AndroidManifest icon bug

> appium-windows-driver@1.5.1 install /usr/local/lib/node_modules/appium/node_modules/appium-windows-driver
> node install-npm.js

Not installing WinAppDriver since did not detect a Windows system

> fsevents@2.0.1 install /usr/local/lib/node_modules/appium/node_modules/fsevents
> [ -f fsevents.node ] || npm run prepare

> heapdump@0.3.9 install /usr/local/lib/node_modules/appium/node_modules/heapdump
> node-gyp rebuild

  CXX(target) Release/obj.target/addon/src/heapdump.o
In file included from ../src/heapdump.cc:17:
../src/compat-inl.h:300:19: error: no member named 'GetCpuProfiler' in 'v8::Isolate'
  return isolate->GetCpuProfiler()->StartProfiling(title, record_samples);
         ~~~~~~~  ^
../src/compat-inl.h:310:19: error: no member named 'GetCpuProfiler' in 'v8::Isolate'
  return isolate->GetCpuProfiler()->StopProfiling(title);
         ~~~~~~~  ^
../src/heapdump.cc:106:23: warning: 'Utf8Value' is deprecated: Use Isolate version [-Wdeprecated-declarations]
    String::Utf8Value filename_string(args[0]);
                      ^
/Users/vsts/.node-gyp/11.5.0/include/node/v8.h:2991:5: note: 'Utf8Value' has been explicitly marked deprecated here
    V8_DEPRECATED("Use Isolate version",
    ^
/Users/vsts/.node-gyp/11.5.0/include/node/v8config.h:326:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated(message)))
                            ^
../src/heapdump.cc:143:9: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
  node::MakeCallback(isolate, isolate->GetCurrentContext()->Global(), callback,
        ^
/Users/vsts/.node-gyp/11.5.0/include/node/node.h:171:1: note: 'MakeCallback' has been explicitly marked deprecated here
NODE_DEPRECATED("Use MakeCallback(..., async_context)",
^
/Users/vsts/.node-gyp/11.5.0/include/node/node.h:91:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
../src/heapdump.cc:157:44: warning: 'Int32Value' is deprecated: Use maybe version [-Wdeprecated-declarations]
  PlatformInit(args.GetIsolate(), args[0]->Int32Value());
                                           ^
/Users/vsts/.node-gyp/11.5.0/include/node/v8.h:2572:3: note: 'Int32Value' has been explicitly marked deprecated here
  V8_DEPRECATED("Use maybe version", int32_t Int32Value() const);
  ^
/Users/vsts/.node-gyp/11.5.0/include/node/v8config.h:326:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated(message)))
                            ^
3 warnings and 2 errors generated.
make: *** [Release/obj.target/addon/src/heapdump.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/vsts/hostedtoolcache/node/11.5.0/x64/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:189:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:254:12)
gyp ERR! System Darwin 17.7.0
gyp ERR! command "/Users/vsts/hostedtoolcache/node/11.5.0/x64/bin/node" "/Users/vsts/hostedtoolcache/node/11.5.0/x64/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/lib/node_modules/appium/node_modules/heapdump
gyp ERR! node -v v11.5.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 

> pre-commit@1.2.2 install /usr/local/lib/node_modules/appium/node_modules/pre-commit
> node install.js


> wd@1.11.1 install /usr/local/lib/node_modules/appium/node_modules/wd
> node scripts/build-browser-scripts


> spawn-sync@1.0.15 postinstall /usr/local/lib/node_modules/appium/node_modules/spawn-sync
> node postinstall

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: heapdump@0.3.9 (node_modules/appium/node_modules/heapdump):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: heapdump@0.3.9 install: `node-gyp rebuild`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1

+ appium@1.11.0
added 1327 packages from 1519 contributors in 73.419s

> wd@1.11.1 install /Users/vsts/agent/2.146.0/work/node_modules/wd
> node scripts/build-browser-scripts

npm WARN saveError ENOENT: no such file or directory, open '/Users/vsts/agent/2.146.0/work/package.json'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN enoent ENOENT: no such file or directory, open '/Users/vsts/agent/2.146.0/work/package.json'
npm WARN work No description
npm WARN work No repository field.
npm WARN work No README data
npm WARN work No license field.

+ wd@1.11.1
added 94 packages from 100 contributors and audited 265 packages in 8.586s
found 4 vulnerabilities (2 moderate, 2 high)
  run `npm audit fix` to fix them, or `npm audit` for details
##[debug]rc:0
##[debug]success:true
##[debug]task result: Succeeded
##[debug]Processed: ##vso[task.complete result=Succeeded;]Bash exited with return code: 0
##[section]Finishing: Setup Appium server

My question if this is OK?

## Environment

* Appium version (or git revision) that exhibits the issue: 1.11.0
* Last Appium version that did not exhibit the issue (if applicable):
* Desktop OS/version used to run Appium: Mac
* Node.js version (unless using Appium.app|exe): 11.5.0
* Npm or Yarn package manager: ???
* Mobile platform/version under test: Android
* Real device or emulator/simulator: emulator
* Appium CLI or Appium.app|exe: CLI

## Details

## Link to Appium logs


## Code To Reproduce Issue [ Good To Have ]

Please remember that with sample code it's easier to reproduce the bug and it's much faster to fix it.

Please git clone https://github.com/appium/appium and from the `sample-code` directory, use one of your favourite languages and sample apps to reproduce the issue.

In case a similar scenario is missing in sample-code, please submit a PR with one of the sample apps provided.

**Comment by KazuCocoa** [comment]
> + appium@1.11.0
> added 1327 packages from 1519 contributors in 73.419s

You can probably call `appium`. It is no problem.
`heapdump` is a debug module to collect heapdump.

If the heapdump's build issue is python version, appium-doctor shows the message, btw.

**Comment by dpgraham** [comment]
We should still investigate this.

**Comment by BoazWarshawsky** [comment]
@KazuCocoa
For some reason my Python version was 2.7 instead of 2.7.15
I solved the issue by installing 2.7.15

From my side we can close the ticket

**Comment by KazuCocoa** [comment]
Okay. Thanks for the result.
We should address `2.7.15` in the doctor, I think.
